sdk/schemaregistry/azure-data-schemaregistry-avro/src/main/java/com/azure/data/schemaregistry/avro/AvroByteDecoder.java

```diff
@@ -0,0 +1,83 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.data.schemaregistry.avro;
+
+import com.azure.core.util.logging.ClientLogger;
+import com.azure.data.schemaregistry.ByteDecoder;
+import com.azure.data.schemaregistry.SerializationException;
+import org.apache.avro.Schema;
+import org.apache.avro.generic.GenericDatumReader;
+import org.apache.avro.io.DatumReader;
+import org.apache.avro.io.DecoderFactory;
+import org.apache.avro.specific.SpecificDatumReader;
+
+import java.io.IOException;
+
+/**
+ * ByteDecoder implementation with all Avro-specific functionality required to deserialize byte arrays
+ * given an Avro schema.
+ */
+public class AvroByteDecoder extends AvroCodec
+        implements ByteDecoder {
+    private final ClientLogger logger = new ClientLogger(AvroByteDecoder.class);
+    private final DecoderFactory decoderFactory = DecoderFactory.get();
+    private final boolean avroSpecificReader;
+
+    /**
+     * Instantiates AvroByteDecoder instance
+     * @param avroSpecificReader flag indicating if attempting to decode as Avro SpecificRecord
+     */
+    public AvroByteDecoder(Boolean avroSpecificReader) {
+        this.avroSpecificReader = avroSpecificReader;
+    }
+
+    /**
+     * @param b byte array containing encoded bytes
+     * @param object schema for Avro reader read - fetched from Azure Schema Registry
+     * @return deserialized object
+     * @throws SerializationException upon deserialization failure
+     */
+    public Object decodeBytes(byte[] b, Object object) throws SerializationException {
+        if (!(object instanceof Schema)) {
+            throw logger.logExceptionAsError(
+                new SerializationException("Attempted to decode with non-Avro schema object in AvroByteDecoder"));
+        }
+        Schema schema = (Schema) object;
+
+        if (schema.getType().equals(Schema.Type.BYTES)) {
+            return b;
+        }
```

Comment:
I think that is the expectation on how to handle byte-only payloads coming back from a message body.  Avro cares about the length of a byte array when working within a struct because it needs to know how long byte arrays are within the data structure so it doesn't read past the end of a byte array.  This specific case is for payloads that consist of nothing but a single byte array.  Avro can just read the entire payload into a buffer and return it to the user to manage.

I am adding a //todo to revisit this.